### PR TITLE
Lock StorageObjectInUseProtection feature gate to default

### PR DESCRIPTION
**Pull request by ikeeip**
#### What type of PR is this?

/kind cleanup
/kind feature

#### What this PR does / why we need it:

Lock `StorageObjectInUseProtection` feature gate to default.

#### Which issue(s) this PR fixes:

Fixes N/A

#### Special notes for your reviewer:

Additional PR to make locking to default first before removing feature gates logic (https://github.com/kubernetes/kubernetes/pull/104903).

#### Does this PR introduce a user-facing change?

```release-note
-  Feature-gate `StorageObjectInUseProtection` has been deprecated and cannot be disabled. It will be completely removed in 1.25
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/features/kube_features.go [hunk 1]
```diff
@@ -756,7 +756,7 @@ var defaultKubernetesFeatureGates = map[featuregate.Feature]featuregate.FeatureS
 	MemoryManager:                                  {Default: true, PreRelease: featuregate.Beta},
 	CPUCFSQuotaPeriod:                              {Default: false, PreRelease: featuregate.Alpha},
 	TopologyManager:                                {Default: true, PreRelease: featuregate.Beta},
-	StorageObjectInUseProtection:                   {Default: true, PreRelease: featuregate.GA},
+	StorageObjectInUseProtection:                   {Default: true, PreRelease: featuregate.GA, LockToDefault: true}, // remove in 1.23
```
Comment (msau42) [inline pkg/features/kube_features.go:759]: Following Kubernetes deprecation policy of 2 releases, this should be 1.25.

**Review by msau42 — COMMENTED** [review]
Can you add a release note to indicate that the feature gate is deprecated?

File: pkg/features/kube_features.go (hunk 1 above)
Comment (ikeeip) [reply pkg/features/kube_features.go]: Done

**Comment by ikeeip** [comment]
> Can you add a release note to indicate that the feature gate is deprecated?

But isn't this enough?

>#### Does this PR introduce a user-facing change?
>
>```release-note
>- The `StorageObjectInUseProtection` graduated to GA in v1.11 and is unconditionally enabled now.
>```

**Comment by msau42** [comment]
I didn't get any response regarding the specific deprecation wording recommendation for the release note, and it doesn't look like we've been doing this for other features, so this looks ok.

/lgtm
/approve

**Comment by sftim** [comment]
Would it be possible to mention the words “feature gate” in the release note here?

**Comment by ikeeip** [comment]
> Would it be possible to mention the words “feature gate” in the release note here?

Yes, fixed.

> See #105474 for an example of wording.

Thanks for the example.

**Pull request merged**